### ISSUE TITLE
Sample tests are broken on master

### ISSUE THREAD

**Issue by JustinBeckwith** [googleapis/nodejs-game-servers#26]
I was looking at this PR, and noticed this error in the sample tests:
https://github.com/googleapis/nodejs-game-servers/pull/25

```

  Game Servers Quickstart Test
(node:319) UnhandledPromiseRejectionWarning: Error: 8 RESOURCE_EXHAUSTED: Quota limit 'RealmCountPerProjectPerRegion' has been exceeded. Limit: 5 in region us-central1.
    at Object.callErrorFromStatus (/tmpfs/src/github/nodejs-game-servers/node_modules/@grpc/grpc-js/build/src/call.js:30:26)
    at Http2CallStream.call.on (/tmpfs/src/github/nodejs-game-servers/node_modules/@grpc/grpc-js/build/src/client.js:96:33)
    at Http2CallStream.emit (events.js:203:15)
    at process.nextTick (/tmpfs/src/github/nodejs-game-servers/node_modules/@grpc/grpc-js/build/src/call-stream.js:100:22)
    at process._tickCallback (internal/process/next_tick.js:61:11)
(node:319) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
(node:319) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
    10) should create a realm
    11) "after all" hook for "should create a realm"
```

Noticed a couple of things wrong here.  

First - we need a `reap` in the `before` handler of the `create_realm` tests.  We should list all available realms, and delete any realm older than an hour.  This will make sure we don't blow up quota, even if tests are cancelled in the middle before the `after` hook cleans up after itself. 

Second - the tests for everything but `create` are just a copy of the `create` tests 🙍‍♂ They are not testing the samples, just testing create over and over.  We should fix that :)

**Comment by telpirion** [comment]
> Second - the tests for everything but create are just a copy of the create tests

I don't follow. It looks like the tests call list, get, delete:

- [list_realms.test.js](https://github.com/googleapis/nodejs-game-servers/blob/master/samples/test/list_realms.test.js)
- [get_realm.test.js](https://github.com/googleapis/nodejs-game-servers/blob/master/samples/test/delete_realm.test.js#L50)
- [delete_realm.test.js](https://github.com/googleapis/nodejs-game-servers/blob/master/samples/test/delete_realm.test.js#L50)

**Comment by JustinBeckwith** [comment]
🤦‍♂ that's totally on me. Turns out those test files are _mostly_ the same.  I got caught looking at the setup code, which ends up being the majority of tests.  

Thinking out loud - what if we put all the tests into a single file, and did the setup/teardown once?  The samples are still split into their own files, but you only need to create the necessary realms once, and reduce the risk of leaking.
